File: builder-pattern/python/src/Phone.py
# -*- coding: utf-8 -*-
"""
@author: jarry
"""

class Phone():
  name = 'PhoneName'
  __name__ = 'Phone'
  screen = [0, 0]
  gpuType = 0

  def __init__(self):
    pass

  def setName(self, name):
    self.name = name

  def getName(self):
    return self.name

  def setScreen(self, size):
    self.screen = size

  def getScreen(self):
    return self.screen

  def setGpuType(self, type):
    self.gpuType = type

  def getGpuType(self):
    return self.gpuType

File: builder-pattern/ts/src/Phone.ts
export class Phone {
  name: string
  __name__: string
  screen: number[]
  gpuType: number
  constructor() {
    this.name = 'PhoneName'
    this.__name__ = 'Phone'
    this.screen = [0, 0]
    this.gpuType = 0
  }

  setName(name: string) {
    this.name = name
  }

  getName() {
    return this.name
  }

  setScreen(size: number[]) {
    this.screen = size
  }

  getScreen() {
    return this.screen
  }

  setGpuType(type: number) {
    this.gpuType = type
  }

  getGpuType() {
    return this.gpuType
  }
}
